build/MdHelper.cs

```diff
@@ -0,0 +1,174 @@
+    /// <returns>A collection of the release notes.</returns>
+    [Pure]
+    public static IEnumerable<string> ExtractChangelogSectionNotes(string changelogFile, string tag = null)
+    {
+        var content = TextTasks.ReadAllLines(changelogFile).Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
+        var sections = GetReleaseSections(content);
+        var section = tag == null
+            ? sections.First(x => x.StartIndex < x.EndIndex)
+            : sections.First(x => x.Caption.EqualsOrdinalIgnoreCase(tag)).NotNull($"Could not find release section for '{tag}'.");
+
+        return content
+            .Skip(section.StartIndex + 1)
+            .Take(section.EndIndex - section.StartIndex);
+    }
+    
+    /// <summary>
+    /// Reads the specified changelog.
+    /// </summary>
+    /// <param name="changelogFile">The path to the changelog file.</param>
+    /// <returns>A <see cref="ChangeLog"/> object to work with the changelog.</returns>
+    [Pure]
+    public static ChangeLog ReadChangelog(string changelogFile)
+    {
+        var releaseNotes = ReadReleaseNotes(changelogFile);
+        var unreleased = releaseNotes.Where(x => x.Unreleased).ToArray();
+
+        if (unreleased.Length > 0)
+        {
+            Assert.True(unreleased.Length == 1, "Changelog should have only one draft section");
+            return new ChangeLog(changelogFile, unreleased.First(), releaseNotes);
+        }
+
+        Assert.True(releaseNotes.Count(x => !x.Unreleased) >= 1, "Changelog should have at lease one released version section");
+        return new ChangeLog(changelogFile, releaseNotes);
+    }
+
+    /// <summary>
+    /// Reads the release notes from the given changelog file and returns the result.
+    /// </summary>
+    /// <param name="changelogFile">The path to the changelog file.</param>
+    /// <returns>A readonly list of the release sections contained in the changelog.</returns>
+    [Pure]
+    public static IReadOnlyList<ReleaseNotes> ReadReleaseNotes(string changelogFile)
+    {
+        var lines = TextTasks.ReadAllLines(changelogFile).ToList();
+        var releaseSections = GetReleaseSections(lines).ToList();
+
+        Assert.True(releaseSections.Any(), "Changelog should have at least one release note section");
+        return releaseSections.Select(Parse).ToList().AsReadOnly();
+
+        ReleaseNotes Parse(ReleaseSection section)
+        {
+            var releaseNotes = lines
+                .Skip(section.StartIndex + 1)
+                .Take(section.EndIndex - section.StartIndex)
+                .ToList()
+                .AsReadOnly();
+
+            return NuGetVersion.TryParse(section.Caption, out var version)
+                ? new ReleaseNotes(version, releaseNotes, section.StartIndex, section.EndIndex)
+                : new ReleaseNotes(releaseNotes, section.StartIndex, section.EndIndex);
+        }
+    }
+    
+    private static IEnumerable<ReleaseSection> GetReleaseSections(List<string> content)
```

Comment:
This method is replaced.
Instead of adhering to the draconian [keep a changelog](https://keepachangelog.com/en/1.0.0/) rule, we instead grab anything between the H2 headers as changelog content.

Reply:
LGTM